src/haddock/clis/restraints/z_surface_restraints.py

```diff
@@ -0,0 +1,468 @@
+        Dictionary holding the various Calpha coordinates for each selection.
+    """
+    # Load structures atom coordinates
+    pdb_atoms = read_structure(pdb_fpath)
+    # Loop over atoms
+    selection_coords: dict[str, list[tuple[float, float, float]]] = {}
+    for _chain, resi, aname, coords in pdb_atoms:
+        # Simplify the probleme to Calpha only
+        if aname != 'CA':
+            continue
+        # Loop over selections
+        for selection, sele_resis in selections.items():
+            # Check if residue of interest
+            if resi in sele_resis:
+                # Cehck if key was created
+                if selection not in selection_coords.keys():
+                    selection_coords[selection] = []
+                # Hold this coordinates
+                selection_coords[selection].append(coords)
+    return selection_coords
+
+
+def compute_barycenter(
+        resi_coords: list[tuple[float, float, float]],
+        ) -> tuple[float, float, float]:
+    """Compute center of mass of multiple resiudes coordinates.
+
+    Parameters
+    ----------
+    resi_coords : list[tuple[float, float, float]]
+        List of Calpha coordinates.
+
+    Returns
+    -------
+    barycenter : tuple[float, float, float]
+        Corrdinates of the center of mass.
+    """
+    # Combine coordinates
+    xs, ys, zs = [], [], []
+    for coords in resi_coords:
+        xs.append(coords[0])
+        ys.append(coords[1])
+        zs.append(coords[2])
+    # Compute average
+    x_avg = sum(xs) / len(xs)
+    y_avg = sum(ys) / len(ys)
+    z_avg = sum(zs) / len(zs)
+    # Return barycenter
+    barycenter = (x_avg, y_avg, z_avg)
+    return barycenter
+
+
+def load_selections(residues_lists: list[str]) -> dict[str, list[int]]:
+    """Split and cast residues from an initial string to list.
+
+    Parameters
+    ----------
+    residues_lists : list[str]
+        List of strings containing coma separated resiudes indices.
+
+    Returns
+    -------
+    selections: dict[str, list[int]]
+        Dictionary of resiudes indices.
+    """
+    selections: dict[str, list[int]] = {}
+    for listid, str_resiudes in enumerate(residues_lists, start=1):
+        resid_indices: list[int] = []
+        selection_key = f"selection_{listid}"
+        for strresid in str_resiudes.split(','):
+            try:
+                resid = int(strresid)
+            except Exception as _e:
+                msg = f"Could not cast residue {strresid} from {selection_key}"
+                logging.warn(msg)
+            else:
+                resid_indices.append(resid)
+        if resid_indices == []:
+            err_msg = f"Not considering {selection_key} as it is empty !"
+            logging.error(err_msg)
+        else:
+            selections[selection_key] = resid_indices
+    return selections
+
+
+def get_z_coords(
+        select_coords: dict[str, list[tuple[float, float, float]]],
+        padding: float = 10,
```

Comment:
I would lower this number

Reply:
It is a parameter now, and set to 5 by default